polling_stations/apps/data_finder/helpers.py

```diff
@@ -118,72 +118,48 @@ def geocode_point_only(self):
         return self.geocode()
 
 
-class AddressBaseGeocoder(BaseGeocoder):
-
-    def format_postcode(self, postcode):
-        # postcodes in AddressBase are in format AA1 1AA
-        # ensure postcode is formatted correctly before we try to query
-        return Postcode(postcode).with_space
-
-    def get_uprns(self, addresses):
-        return [a.uprn for a in addresses]
-
-    def get_codes(self, uprns):
-        addresses = Onsud.objects.filter(uprn__in=uprns)
-
-        if len(addresses) == 0:
-            # No records in the ONSUD table were found for the given UPRNs
-            # because...reasons
-            raise CodesNotFoundException('Found no records in ONSUD for supplied UPRNs')
-
-        if len(addresses) != len(uprns):
-            # For the moment I'm going to do nothing about this, but lets
-            # leave this condition here to make that decision explicit
-            # TODO: maybe we should actually do....something else??
-            pass
-
-        council_ids = set([a.lad for a in addresses])
-
-        # assemble list of codes
-        gss_codes = set()
-        for address in addresses:
-            extra_codes = [address.cty, address.lad, address.ctry, address.rgn, address.eer]
-            for code in extra_codes:
-                gss_codes.add(code)
+class AddressBaseGeocoderAdapter(BaseGeocoder):
+    """
+    For the moment we need an adapter clas to sit between
+    uk_geo_utils.geocoders.AddressBaseGeocoder
+    and the calling code which wraps up output from
+    uk_geo_utils.geocoders.AddressBaseGeocoder
+    in the data structure that the calling code is expecting.
 
-        if len(council_ids) == 1:
-            # all the uprns supplied are in the same local authority
-            council_gss = list(council_ids)[0]
-        else:
-            # the urpns supplied are in multiple local authorities
-            raise MultipleCouncilsException('Postcode %s covers UPRNs in more than one local authority' % (self.postcode))
-
-        return {
-            'council_gss': council_gss,
-            'gss_codes': list(gss_codes)
-        }
+    Somewhere later in the refactoring process,
+    we can probably remove this abstraction layer
+    """
 
     def geocode(self):
-        addresses = Address.objects.filter(postcode=self.postcode)
-        if not addresses:
-            raise ObjectDoesNotExist('No addresses found for postcode %s' % (self.postcode))
+        geocoder = AddressBaseGeocoder(self.postcode)
+        centre = geocoder.centroid
+
+        try:
+            lad = geocoder.get_code('lad')
+        except MultipleCodesException as e:
+            # re-raise as a more specific MultipleCouncilsException
+            # because that is what the calling code expects to handle
+            raise MultipleCouncilsException(str(e))
+
+        codes = [
```

Comment:
Remember that the entire `data_finder.helpers.AddressBaseGeocoderAdapter` class is just a temporary 'compatibility' layer to avoid changing too much at once. Given that purpose, it is correct for this to be a list because the old `data_finder.helpers.AddressBaseGeocoder` that the `AddressBaseGeocoderAdapter` 'mimics' represents this as a list. This code will be replaced but this is just an 'intermediate' step which helps break this big task down into smaller chunks.